tests/scripts/generate_test_driver.py

```diff
@@ -0,0 +1,181 @@
+
+from config import TFPSACryptoConfig
+
+EXCLUDE_FILES = {
+    "asn1*",
+    "base64*",
+    "crypto_builtin_key_derivation.h",
+    "lmots*",
+    "lms.c",
+    "md.c",
+    "memory_buffer_alloc.c",
+    "nist_kw.c",
+    "pem.c",
+    "pk*",
+    "platform*",
+    "threading*",
+}
+
+IDENTIFIER_PREFIXES = {
+    "MBEDTLS_",
+    "PSA_",
+    "TF_PSA_CRYPTO_",
+    "mbedtls_",
+    "psa_",
+    "tf_psa_crypto_",
+}
+
+def iter_code_files(root: Path) -> Iterable[Path]:
+    """
+    Iterate over all "*.c" and "*.h" files found recursively under the `include`
+    and `src` subdirectories of `root`.
+    """
+    for directory in ("include", "src"):
+        directory_path = root / directory
+        for ext in (".c", ".h"):
+            yield from directory_path.rglob(f"*{ext}")
+
+def get_src_relpaths(builtin: Path, exclude_files: Set[str]) -> List[Path]:
+    """
+    Return the relative paths of all *.c and *.h files under `builtin`,
+    excluding those whose names match any of the patterns in `exclude_files`.
+
+    The returned paths are relative to `builtin`.
+    """
+
+    out = []
+    for file in iter_code_files(builtin):
+        if not any(fnmatch(file.name, pattern) for pattern in exclude_files):
+            out.append(file.relative_to(builtin))
+    out.sort()
+    return out
+
+def get_dst_relpaths(src_relpaths: List[Path], driver: str) -> List[Path]:
+    """
+    Return the relative paths of the *.c and *.h files generated by the script.
+
+    These paths are the same as in `src_relpaths`, except that occurrences of
+    `mbedtls` in `include/mbedtls/...` paths are replaced with `driver`.
+
+    The returned paths are relative to `dst_dir`.
+    """
+
+    out = []
+    for path in src_relpaths:
+        parts = list(path.parts)
+        if parts[0] == "include" and parts[1] == "mbedtls":
+            parts[1] = driver
+        out.append(Path(*parts))
+
+    return out
+
+def get_external_identifiers() -> Set[str]:
+    """
+    Get from public and core headers the identifiers that the driver built-in
+    code may reference but does not define.
+    """
+    directories = ("core", "include")
+    files = itertools.chain.from_iterable(Path(directory).rglob("*.h") \
+                                          for directory in directories)
+    identifiers = set()
+    for file in files:
+        identifiers.update(test_driver.run_ctags(file))
+
+    # MBEDTLS_PRIVATE is returned as a prototype by ctags when used in
+    # structure members. Just remove it.
+    identifiers.remove("MBEDTLS_PRIVATE")
+
+    # ctags ignores the configuration options that are commented in
+    # crypto_config.h. Ensure we have all of them.
+    identifiers |= set(TFPSACryptoConfig().settings)
+
+    return identifiers
+
+def main():
+    """
+    Main function of this program
+    """
+    parser = test_driver.get_parsearg_base()
+    parser.add_argument('--list-vars-for-cmake', nargs="?", const="__AUTO__", metavar="FILE",
```

Comment:
Please document that this is done _instead_ of creating a test driver tree, not in addition.